src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs

```diff
@@ -499,62 +499,63 @@ select game
             {
                 if (GameValidation.AreSameTeamsInGames(game, newGame))
                 {
-                    if (GameValidation.IsFreeDayGame(newGame))
-                    {
-                        if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff)
-                        {
-                            throw new ArgumentException(
-                            Resources
-                            .SameFreeDayGameInRound);
-                        }
-                        else
-                        {
-                            throw new ArgumentException(
-                                string.Format(Resources.SameTeamInRound, game.HomeTeamName));
-                        }
-                    }
-                    else
-                    {
-                        throw new ArgumentException(
-                           string.Format(
-                           Resources.SameGameInRound,
-                           game.HomeTeamName,
-                           game.AwayTeamName,
-                           game.Round.ToString()));
-                    }
+                    ValidateAreSameTeamsInGames(game, newGame, tournamentScheduleInfo);
                 }
                 else if (GameValidation.IsTheSameTeamInTwoGames(game, newGame))
                 {
-                    if (GameValidation.IsFreeDayGame(newGame))
-                    {
-                        if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff
-                            && game.HomeTeamId != newGame.HomeTeamId
-                            && game.AwayTeamId != newGame.HomeTeamId)
-                        {
-                            throw new ArgumentException(
-                                Resources.SameFreeDayGameInRound);
-                        }
-                    }
-                    else
-                    {
-                        string oppositeTeam;
+                    ValidateIsTheSameTeamInTwoGames(game, newGame, tournamentScheduleInfo);
+                }
+            }
+        }
 
-                        if (game.HomeTeamId == newGame.HomeTeamId
-                            || game.HomeTeamId == newGame.AwayTeamId)
-                        {
-                            oppositeTeam = game.HomeTeamName;
-                        }
-                        else
-                        {
-                            oppositeTeam = game.AwayTeamName;
-                        }
+        private static void ValidateAreSameTeamsInGames(GameResultDto game, Game newGame,
+           TournamentScheduleDto tournamentScheduleInfo)
+        {
+            if (GameValidation.IsFreeDayGame(newGame))
+            {
+                if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff)
```

Comment:
Build a message then throw an error

Reply:
but it will be a message ("Free day game has already been scheduled in this round"); or I bad understand you, I did not make a logic change here.